txnsync/bloomFilter.go

```diff
@@ -136,7 +139,7 @@ func filterFactoryXor32(numEntries int, s *syncState) (filter bloom.GenericFilte
 
 var filterFactory func(int, *syncState) (filter bloom.GenericFilter, filterType bloomFilterType) = filterFactoryXor32
 
-func (s *syncState) makeBloomFilter(encodingParams requestParams, txnGroups []pooldata.SignedTxGroup, excludeTransactions *transactionCache, hintPrevBloomFilter *bloomFilter) (result bloomFilter) {
+func (s *syncState) makeBloomFilter(encodingParams requestParams, txnGroups []pooldata.SignedTxGroup, excludeTransactions *transactionCache, hintPrevBloomFilter *bloomFilter, minGroupCounter uint64) (result bloomFilter) {
```

Comment:
The changes that you've made here could be done by the caller, which would do them more efficiently : 
instead of comparing each of the groups in txnGroups to minGroupCounter, the caller should use sort.Search on the txnGroups. The entries in the txnGroups are ordered by their GroupCounter, which would allow you to perform a binary search. Then, sub-slice the txnGroups and pass it to `makeBloomFilter`